source/blog/2017-10-16-fall-in-for-finagle.md

```diff
@@ -0,0 +1,144 @@
+
+### Operability
+
+We’ll be open-sourcing the work done by our summer intern [@McKardah](https://twitter.com/McKardah) to wire
+Twitter Futures into [IDEA’s async
+stacktraces](https://blog.jetbrains.com/idea/2017/02/intellij-idea-2017-1-eap-extends-debugger-with-async-stacktraces/).
+
+We’ve completed the internal rollout of [edge-triggered native
+transport](https://github.com/twitter/finagle/search?utf8=%E2%9C%93&q=nativeEpoll&type=) using Netty’s JNI edge-triggered transport, instead of using the JDK’s NIO and are working to make it available to configure externally.
+
+#### Security
+
+Secure Mux (smux). We've added a
+[STARTTLS](https://en.wikipedia.org/wiki/Opportunistic_TLS) command into the
+Mux protocol in order to support opportunistic TLS for ThriftMux.
+
+#### Logging Improvements
+
+We've been working hard to move
+[TwitterServer](https://github.com/twitter/twitter-server) to the [SLF4J
+API](https://) and this should be merging in the near future. Stay tuned for
+updates and documentation on the changes.
+
+#### Scrooge Updates
+
+We've been making lots of changes to improve the usability and performance of
+Scrooge and Scrooge-generated code. This includes a set of refactorings to modernize the generated code so that it’s written in terms of Filters and Services!
+
+This will allow us to address some long standing issues like server-side response classification and correctly stating on serialization/deserialization failures.
+
+Additionally, we going to try to update to libthrift 0.10.0 from 0.5.0. This is still exploratory and we’re working to understand the ramifications.
+
+#### Bringing Backup Requests Back
+
+Finagle has had a
+[BackupRequestFilter](https://github.com/twitter/finagle/blob/develop/finagle-exp/src/main/scala/com/twitter/finagle/exp/BackupRequestFilter.scala)
+since [late
+2012](https://github.com/twitter/finagle/commit/526e508579309711a9c56007eff2a783826d331c)
+and evolved into its current form in [early
+2013](https://github.com/twitter/finagle/commit/8ca92593aac1d3f5bd4741d6b8e6fcf26053dc44).
+The filter has a variety of what we would consider flaws (too many
+configuration knobs, less than ideal memory characteristics in many cases,
+lack of integration into the Stack-APIs and ClientBuilder to name a few) which
+has relegated it to Finagle's experimental module.
+
+We're investigating what it would take to give users a single configuration
+variable -- ideally the percentage of additional requests to send -- to
+configure sending a backup request to another server.
+
+#### ‘Push’ based API for Protocols
+
+Modern multiplexing protocols are characteristically very stateful and event
+driven making them challenging to implement with the current ‘pull’ based
```

Comment:
"... driving _which makes_ them challenging ..." seems to flow better, at least for me.